src/components/SetupRedirectButton.tsx

```
@@ -0,0 +1,56 @@
+ */
+export default function SetupRedirectButton({ 
+  className = "btn-primary", 
+  children = "Complete seu perfil" 
+}: SetupRedirectButtonProps) {
+  const { canAccessSetup, redirectToDashboard, isLoading } = useSetupGuard();
+
+  // Se ainda está carregando, mostra loading
+  if (isLoading) {
+    return (
+      <button className={className} disabled>
+        <div className="animate-spin rounded-full h-4 w-4 border-b-2 border-white mx-auto"></div>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)